.github/workflows/main.yml

```diff
@@ -7,9 +7,10 @@ jobs:
     name: Check
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v1
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
         with:
+          distribution: 'zulu'
```

Comment:
Why zulu and not temurin?

Reply:
We are using the same one as before:  
https://github.com/actions/setup-java/blob/main/docs/switching-to-v2.md

You can change it if you want:  
https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions

Reply:
Ahh okay, do you have a preference? I think temurin is more open, as it's maintained  eclipse?

Reply:
Based on https://whichjdk.com/, temurin sounds good

Reply:
I mostly use Zulu or SapMachine, but only because they just work. I never bothered to change it for my projects.

Reply:
Okay, then let's go with temurin, if the build succeeds (:.